src/ck-core/cklocation.C

```diff
@@ -2498,6 +2537,19 @@ void CkLocMgr::processAfterActiveRgetsCompleted(CmiUInt8 id)
   }
 }
 
+void CkLocMgr::deliverAllBufferedMsgs(CmiUInt8 id)
```

Comment:
Once messaging moves to CkArray this goes away/is significantly simplified. But the distinction comes from the reasons for buffering. Sometimes messages are buffered because we are still dealing with array indices, and some are buffered after we leave index-land and move to ID-only-town.